src/simplewallet/simplewallet.cpp

```diff
@@ -2227,6 +2226,7 @@ bool simple_wallet::transfer_main(int transfer_type, const std::vector<std::stri
     else while (!ptx_vector.empty())
     {
       auto & ptx = ptx_vector.back();
+      ptx.is_onetime = is_onetime;
```

Comment:
This should be irrelevant to get_payment_id. Functions like create_transactions_2 fill up the ptx set, and should set the is_onetime from the passed parameter. This is done just before in the timeline, so get_payment_id does not need to change.
add_unconfirmed_tx saves data about an outgoing tx that cannot be recovered from the blockchain (for examaple, destination addresses). It could also save short payment_id, since this is not decryptable after the fact by the sender (except if the sender is the receiver).